src/audio/mixin_mixout/mixin_mixout.c

```diff
@@ -423,7 +423,11 @@ static int mixin_process(struct processing_module *mod,
 		 * performance problems with processing data on time in mixout.
 		 */
 		start_frame = mixout_data->pending_frames[source_index];
-		assert(sinks_free_frames >= start_frame);
+		if (start_frame > frames_to_copy) {
+			comp_warn(dev, "mixin copy: no space, need %u, space for %u",
+				  start_frame, frames_to_copy);
+			frames_to_copy = start_frame;
```

Comment:
mmh, wait, why truncate? `start_frame` is what was in the buffer before this processing run. And `frames_to_copy` contains that plus new data from this run. Why not attempt to send it all?

Reply:
@lyakh Sorry, this was an old commit cherry-picked here for testing. The merged version of this patch did not truncate.